libyul/AsmParser.cpp

```diff
@@ -249,6 +274,8 @@ ForLoop Parser::parseForLoop()
 	forLoop.pre = parseBlock();
```

Comment:
I think we should set `m_insideForLoopBody` to false here already.

Reply:
I still think we should do this.

Reply:
now i realize your motivation behind. if you're having nested for loops and then you could use break inside pre/post of the inner loop statement. that of course must be prohibited. thx.